From 2cb65bc75b93529e4029155fd1c0f31065f6121d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <cynerd@email.cz>
Date: Fri, 20 Nov 2020 10:33:22 +0100
Subject: [PATCH] approvals: fix exception on approval required

The autorun.auto_approve_time() returns None in case of approvals are
required. Instead of checking if it is not None we checked old condition
and that if it is larger than 0.

This was casing update blockage in case approvals were configured as
required.
---
 CHANGELOG.md           | 3 +++
 svupdater/approvals.py | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG.md b/CHANGELOG.md
index 70b299a..f477138 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Fixed
+- exception raised when "approvals needed" was configured there was pending
+  approval
 
 
 ## [1.3.1] - 2020-05-06
diff --git a/svupdater/approvals.py b/svupdater/approvals.py
index ae842e4..88e966b 100644
--- a/svupdater/approvals.py
+++ b/svupdater/approvals.py
@@ -154,7 +154,7 @@ def _approved():
     with open(const.APPROVALS_STAT_FILE, 'r') as file:
         cols = file.readline().split(' ')
         auto_grant_time = autorun.auto_approve_time()
-        if cols[1].strip() == 'granted' or (auto_grant_time > 0 and int(cols[2]) < (time.time() - (auto_grant_time * 3600))):
+        if cols[1].strip() == 'granted' or (auto_grant_time and int(cols[2]) < (time.time() - (auto_grant_time * 3600))):
             return cols[0]
         return None
 
-- 
2.29.2

